tests/models/seamless_m4t_v2/test_modeling_seamless_m4t_v2.py

```diff
@@ -837,6 +846,21 @@ def factory_generation_speech_test(self, model, inputs):
         output = model.generate(**inputs)
         return output
 
+    def test_generation_languages(self):
+        config, input_text_rus = self.prepare_text_input(is_rus=True)
+
+        model = SeamlessM4Tv2Model(config=config)
+        self.update_generation(model)
+        model.to(torch_device)
+        model.eval()
+
+        # make sure that generating speech, with a language that is only supported for text translation, raises error
+        with self.assertRaises(ValueError):
+            model.generate(**input_text_rus)
+
+        # make sure that generating text only works
+        model.generate(**input_text_rus, generate_speech=False)
```

Comment:
We should also make sure it works in both cases for a language supported for all tasks